src/main/java/exercises/nguonchhay/OopExercise.java

```diff
@@ -0,0 +1,315 @@
+		return state;
+	}
+	
+	public void setState(long state) {
+		this.state = state;
+	}
+	
+	public long getVersion() {
+		return version;
+	}
+	
+	public void setVersion(long version) {
+		this.version = version;
+	}
+}
+
+/**
+ * Person Class
+ */
+class Person extends DCommon {
+	protected Date birthDate;
+	protected String lastName;
+	protected String firstName;
+	
+	public Person() {}
+	
+	public Person(String lastName, String firstName, Date birthDate) {
+		setLastName(lastName);
+		setFirstName(firstName);
+		setBirthDate(birthDate);
+	}
+	
+	public Person(long id, String lastName, String firstName, Date birthDate, Date createdDate, Date updatedDate, String createdBy, String updatedBy, long state, long version) {
+		super(id, createdDate, updatedDate, createdBy, updatedBy, state, version);
+		setLastName(lastName);
+		setFirstName(firstName);
+		setBirthDate(birthDate);
+	}
+	
+	public Date getBirthDate() {
+		return birthDate;
+	}
+	
+	public void setBirthDate(Date birthDate) {
+		this.birthDate = birthDate;
+	}
+	
+	public String getBirthDateString() {
+		if (null == birthDate) {
+			return "";
+		}
+		return birthDate.toString();
+	}
+	
+	public String getLastName() {
+		return lastName;
+	}
+	
+	public void setLastName(String lastName) {
+		this.lastName = lastName;
+	}
+	
+	public String getFirstName() {
+		return firstName;
+	}
+	
+	public void setFirstName(String firstName) {
+		this.firstName = firstName;
+	}
+	
+	public String getFullname() {
+		return new StringBuffer(lastName + " " + firstName).toString().toUpperCase();
+	}
+	
+	public String toString() {
+		return super.getId() + " " + getFullname() + " " + getBirthDateString() + " " + super.getCreateDateString() + " " + super.getUpdatedDateString().toString()
+				+ " " + super.getCreatedBy() + " " + super.getUpdatedBy() + " " + super.getState() + " " + super.getVersion();
+	}
+}
+
+/**
+ * Employee Class
+ */
+class Employee extends Person implements View {
+	
+	protected double salary;
+	protected double years;
+	
+	public Employee() {}
+	
+	public Employee(double salary, double years) {
+		setSalary(salary);
+		setYears(years);
+	}
+	
+	public Employee(String lastName, String firstName, Date birthDate, double salary, double years) {
+		super(lastName, firstName, birthDate);
+		setSalary(salary);
+		setYears(years);
+	}
+	
+	public Employee(long id, String lastName, String firstName, Date birthDate, double salary, double years, Date createdDate,
+			Date updatedDate, String createdBy, String updatedBy, long state, long version) {
+		super(id, lastName, firstName, birthDate, createdDate, updatedDate, createdBy, updatedBy, state, version);
+		setSalary(salary);
+		setYears(years);
+	}
+	
+	public double getSalary() {
+		return salary;
+	}
+
+	public void setSalary(double salary) {
+		this.salary = salary;
+	}
+
+	public double getYears() {
+		return years;
+	}
+
+	public void setYears(double years) {
+		this.years = years;
+	}
+
+	@Override
+	public String getDataView() {
```

Comment:
public String getDataView() {
        final SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy/MM/dd");
        final String line = System.lineSeparator();
        StringBuilder sb = new StringBuilder();
        sb.append("id :").append(getId()).append(line);
        sb.append("FirstName : ").append(getFirstName()).append(line);
        sb.append("LastName : ").append(getLastName()).append(line);
        if (getBirthDate() != null) {
            sb.append("Date Of Birth : " ).append(dateFormat.format(getBirthDate())).append(line);
        }
        sb.append("salary : ").append(getSalary()).append(line);
        sb.append("working in company of years :").append(getYears());
       
        return sb.toString();
    }
}